### DISCUSSION TITLE
Liczenie kosztów on-premise vs Azure VMs. (cała serwerownia)

### DISCUSSION THREAD

**Discussion by phaate** [AzureCommunityPL/group#270] [Pomoc]
Cześć,

Potrzebuję wyliczyć koszt wypchnięcia jednej z serwerowni do Az. Wiadomo, projekty przychodzą i odchodzą więc tu policzę dla obecnego wykorzystania.
Jak liczycie w takich przypadkach koszty infry? Total Cost of Ownership, jakieś customowe kalkulatory czy zwykły azure calc i liczenie dla każdej VMki?

Niestety projekty realizowane przez firmę to głównie c, cpp itd i firma "przyzwyczajona" jest do VMek, niespecjalnie jest co wykorzystać na Az żebym mógł ich przerzucić na pasy, sasy.

Rozważane jest kupienie sprzętu i kolokacja lub wystawienie całości na Az.

Jakieś pomysły, rady jak się zabrać do tematu żeby wyszło taniej/lepiej u niebieskich ?:>

**Comment by kamil-mrzyglod** [comment]
Ostatecznie to pewnie będzie wypadkowa TCO / Azure Calculator / Obliczeń w Excelu. Ciężko wyliczyć cokolwiek bliskiego realnemu zużyciu na jednym narzędziu bo albo czegoś tam nie ma, albo nie do końca jesteś w stanie wykazać poprawną charakterystykę. 

Ja bym pewnie wyszedł z TCO(https://azure.microsoft.com/en-us/pricing/tco/calculator/), popatrzył trochę na kalkulator od Azure(https://azure.microsoft.com/en-us/pricing/calculator/) i postarał się wyliczyć z grubsza ile same VMki(w sensie compute) wyniesie. Potem jeszcze pozostają rzeczy jak:

- Hybrid Benefit jeśli masz u siebie Windows(https://docs.microsoft.com/en-us/azure/virtual-machines/windows/hybrid-use-benefit-licensing#:~:text=%20There%20are%20few%20ways%20to%20use%20Windows,Benefit%20or%20pay%20on-demand%20cost%20for...%20More%20)
- Reserved Instance(https://azure.microsoft.com/en-us/pricing/reserved-vm-instances/)
- do rozważenia Spot Instances(https://docs.microsoft.com/en-us/azure/virtual-machines/spot-vms)

Jak masz tego trochę to ogarnięty partner / Enterprise Agreement. Ostatecznie może i będzie lepiej ale z tym taniej to się nie podpiszę ;)

EDIT: A no jeszcze pozostaje doliczyć koszt za storage + ruch wychodzący z DC jeśli takowy będzie.

**Comment by KamilWiecek** [comment]
Cześć, 

Możesz zaczać od [Azure Migrate](https://docs.microsoft.com/en-us/azure/migrate/how-to-create-assessment). 

Niejedna analiza wykonana w podobny do Twojego pomysłu sposób, zakończyła się wnioskiem, że *chmura jest droższa* od on-premises. Może to być prawdą, zwłaszcza jeżeli nie korzystamy w pełni z jej możliwości. Jeżeli skala jest duża, to samo zrobienie inventory tego co jest teraz, może zaowocować redukcjami zasobów pod poszczególne workloady. Do tej pory mogły być one przewymiarowane, żeby "potem nie trzeba było prosić o więcej". 

Spróbuj uwzględnić w swoich kalkulacjach:
- obecną utlizację zasobów (CPU/ RAM)
- [hybrid benefit](https://azure.microsoft.com/en-us/pricing/hybrid-benefit/)
- [reserved instances](https://azure.microsoft.com/en-us/pricing/reserved-vm-instances/) i/ lub wyłączanie maszyn, wtedy płacisz tylko za storage
- [odpowiednie typy subskrypcji](https://azure.microsoft.com/en-us/pricing/dev-test/)
- skalowanie on-demand
- zzyski z automatyzacji, która jest po prostu prostsza.

Wiele decyzji i potencjalnych oszczędności zależy od specyfiki projektów i workloadów, które wylądują na tych maszynach. 
Warto zastanowić się, jak Azure wpłynie na koszty operations, które teraz robicie - configration management, updates, backup, monitring, skany security, itd. Jeżeli coś uprości, to wpłynie to na koszty. Chmura pozwoli Ci też rozliczyć łatwiej te koszty pomiędzy projekty/ zespoły.

**Comment by piotrgwiazda** [comment]
Wirtualna serwerownia to nie chmura. To tylko hosting i to dość drogi. Bez
spojrzenia na replatforming w kierunku cloud native nie spodziewałbym się
szalonego uzysku.

pon., 26 kwi 2021, 15:13 użytkownik Kamil ***@***.***>
napisał:

**Reply by kamil-mrzyglod** [reply]
> Wirtualna serwerownia to nie chmura. To tylko hosting i to dość drogi. Bez spojrzenia na replatforming w kierunku cloud native nie spodziewałbym się szalonego uzysku. 

Szalonego na pewno nie jeśli jakikolwiek będzie w ogóle przy lift & shift. Cloud native to dobry kierunek, ale też nie do brania za pewnik - MS niezły makaron nawija przy niektórych usługach i przy większej skali połowa kosztów to element, którego nawet w cenniku nie widziałeś. To jest zresztą prawdziwe dla większości dostawców chmurowych, koniec końców to nie jest działalność charytatywna. 

A tak w ogóle wracając do tematu @phaate-ls  - masz cały Cloud Adoption Framework(https://azure.microsoft.com/en-us/cloud-adoption-framework/#cloud-adoption-journey) gdzie masz trochę benchmarków i dobrych praktyk jak podejść do migracji i jakie kroki podjąć dalej. Jak nie masz dobrego partnera i/lub doświadczenia to prędzej urwiesz sobie łapę niż zrobisz to dobrze. Tutaj jest za dużo ruchomych elementów i jeśli długofalowym celem nie jest modernizacja stacku to nie wiem czy bym wchodził w chmurę(bo argument tanio/drogo to żaden argument, w 9/10 przypadków chmura przegrywa).

**Reply by phaate** [reply]
@kamil-mrzyglod szczerze, nie migrowałem nigdy dla takiej skali i takiego stacku. Bądź co bądź firma robi większość w c, c++, linux kernel development itd gdzie VMka jest remedium na całe zło i reorganizacja ma się odbyć głównie w obrębie "data warehouse" i projektów z nim związanych ale zespół... jest daleko w tyle.

@piotrgwiazda Nawet niespecjalnie wiem co miałbym takiemu linux kernel devowi zaproponować więc nie liczyłbym na przejście w pasy czy sasy. Firma klepie projekty dla silicon, automotive itd.

@kwiecek kompletnie wyleciał mi z głowy azure migrate, dzięki.
Z informacji, które mam to nie mogę ingerować w to co devowie sobie zażyczą. Proszą o 16core 32gb ram i PM to zaakceptuje to mają dostać (to nic, że 23h z 24h leży odłogiem). Jedyną opcją skalowania to wertykalnie. Nawet projekty, które są stworzone do skalowania horyzontalnego w czasie developmentu dostają 3 vmki i nie wolno dotykać. Dla takich liczę 3h tier wyższy + power automate + niższy tier kiedy vmka leży i nic nie robi.

Zacząłem standardowo z CAFem, projektując sieć i subskrypcje + backup. Schodki się zaczęły przy VMkach i to co w nich jest i dlatego podpytuje czy moja metodologia nie jest przypadkiem zła... well, każdemu zdaża się zwątpić w swoje wyliczenia. 
Jedyne co realnie mogłem zaproponować to "zbicie" bd w managed instance i apki na app serwisach / vmkach + całe raportowanie na databricksach, data factory i data lake'u.

**Comment by piotrgwiazda** [comment]
Sporo rzeczy daje się konteneryzować i odpalać na AKS. Róznica w obsłudze
jest kolosalna. A kubernetes to nowy bash - linux dev musi to umieć w 2021.

Co do baz - a dlaczego MI? To ostateczność tylko i wyłącznie dla tych baz
do już naprawdę nie ma innej opcji bo tak jest w nich namieszane, że nie da
się odkręcić.

pon., 26 kwi 2021, 21:30 użytkownik phaate-ls ***@***.***>
napisał:

**Reply by phaate** [reply]
@piotrgwiazda Chwilę się zastanawiałem nad tym i MI najlepiej mi się wpisało w to co aktualnie stoi. Jedna duża baza + testówki + małe bazy dla devów. Żadnych shardingów, failoverów i innych skomplikowanych konfiguracji. Jeden SSIS ale wiadomo, tego na Az nie ma więc całą analitykę/raportowanie trzeba będzie przerobić.

Brałem też pod uwagę po prostu VMkę z BYOL i elastic pool lub jeden duży PaS na główną bazę + elastic pool na mniejsze. Możesz rozwinąć co dokładnie jest nie tak z MI? Aż takiego doświadczenia z MI nie mam więc ciekawy jestem opinii kogoś bardziej doświadczonego.

**Reply by jmcee1** [reply]
[SSIS też można przenieść w całości](https://docs.microsoft.com/en-us/azure/data-factory/concepts-integration-runtime#azure-ssis-integration-runtime) tylko takie wypychanie nie ma szans żeby wyszło taniej na kalkulatorze.
SQL MI nie pozwala się zatrzymywać na żądanie czyli ewentualnie możnaby ją brać pod uwage jako główną bazę. Tylko tak, jak pisze Piotr, jeśli nie mamy do czynienia z dopasowanym MSSQL to czemu nie skorzystać z rozwiązań stworzonych dla chmury, prostszych w obsłudze i potencjalnie tańszych?

**Reply by phaate** [reply]
MI zdawal się lepszym wyborem na start przy takim lift and shift. SQL MI ma mniej "gwiazdek" i roznic miedzy standardowym SQL Serverem na VMce. 

I tak wszystko wyjdzie na testach czy zwykly elastic pool da rade czy wolą managed instance ale dzięki za uwagi :)

**Comment by piotrgwiazda** [comment]
Azure SQL MI wstaje około 6 godzin (Azure SQL DB parę minut) i nie ma
wsparcia dla Terraform zbyt dobrego. Masz tam masę pokręteł, których nie
musisz oglądać w SQL DB. Model sieciowy wyglada inaczej. Pricing wygląda
inaczej.

wt., 27 kwi 2021, 09:53 użytkownik phaate-ls ***@***.***>
napisał:

**Reply by miqm** [reply]
przy migracjach baz idź w MI. sporo case studies jeśli chodzi o migracje bardzo dużych serwerów SQL on-prem do Azure właśnie opierało się na na MI po stronie Azure.  

Na Terraformie świat się nie kończy :)

**Comment by phaate** [comment]
Dzięki wszystkim za sugestie i rady. 
Trochę ściągnąłem przykładów i pomysłów z webinara Kamila więc dzięki @kwiecek .
Przedstawilem migracje ssis do az + bazy do wyboru elastic lub mi.

VMki to dosc ciezki temat i bez skalowania/automatyzacji az mnie przerazily koszty jak zrobilem to bez optymalizacji na azure calcu.

**Reply by kamil-mrzyglod** [reply]
To jest standardowa reakcja ;) Jak robiłem kiedyś webinar o Azure i pokazyłem VMki na kalkulatorze to od razu zaczęły się głosy "Ło Panie, to ja na takim OVH w tej cenie postawię 3 maszyny i jeszczę za kawę zapłace". Kto może to z tego modelu ucieka bo koszty rosną wykładniczo do skali.

**Comment by krukowskid** [comment]
Jeżeli masz trochę towaru do wypchnięcia i zdecydujesz się na Azure to warto odezwać się do MS. Jest szansa na dostanie środków w Azure na start + pomocy w migracji przez wyznaczonego przez MS partnera.
